sound/soc/sof/sof-priv.h

```diff
@@ -238,6 +238,10 @@ struct snd_sof_dsp_ops {
 	/* pcm ack */
 	int (*pcm_ack)(struct snd_sof_dev *sdev, struct snd_pcm_substream *substream); /* optional */
 
+	/* pcm delay */
+	int (*pcm_delay)(struct snd_sof_dev *sdev, struct snd_soc_component *component,
+			 struct snd_pcm_substream *substream); /* optional */
```

Comment:
By looking at this, it is not clear what the implementation should be doing. Delay something or return the delay as integer (opposed to `snd_pcm_sframes_t`).
The only place from where this is called does not check the return value...

Imho this callback should return the delay value. But the question is what delay the dsp_ops level of delay should be returning?

I think we are mixing the layers here.. The MTL implementation just reads the DMA position on the DAI side, no?

Reply:
we will calculate delay in ipc4 core layer and need to get hw info from dsp layer.

Reply:
I had missed this and confused with the IPC4 core layer delay(). But yeah, I think @ujfalusi  has a point, this DSP op is not calculating the delay, so the name is not good. Maybe "update_stream_position()"? This would update "stream->posn" in a DSP specific way. OTOH, then why don't we call this elsewhere? Other option is for this function to return "stream->posn" .

Reply:
good idea. Thanks!